### DISCUSSION TITLE
Requesting an example of a python plugin

### DISCUSSION THREAD

**Discussion by photkey** [Flow-Launcher/Flow.Launcher#842] [General]
Have already seen the official example: https://github.com/deefrawley/Flow.Launcher.Plugin.Currency and Search-MDI, a python plugin (without using the official template), but still can't achieve the functionality I want.
I hope to provide an example to demonstrate it, with the simplest implementation, without any special case handling.

Hopefully the example will show a function like this.
Enter a keyword, and the first item in the menu is Yes open "https://s.taobao.com/search?q={q}" from a browser.
The second item is to run the function open_webs below. (just for the sake of example, it does not represent the actual function I want)

```
def open_webs(q:str):
    q=q.replace(' ','')
    urls = [
        'http://search.jd.com/Search?keyword={}&enc=utf-8'.format(q),
        'https://s.taobao.com/search?q={}'.format(q),
    ]
    for url in urls:
        webbrowser.open(url)
```

**Comment by jjw24** [comment]
So you want to return two results from a query? One result is open browser of that url and second result is to run that function? Is this what you are trying to achieve?

**Reply by photkey** [reply]
Yes, thank you very much!
Press Enter to execute the corresponding option, or Alt+1/2 to execute the corresponding option, this part should not need to be written in the code, right?

**Reply by jjw24** [reply]
>Press Enter to execute the corresponding option, or Alt+1/2 to execute the corresponding option, this part should not need to be written in the code, right?

No, dont need to worry about this, the main app when take care of it.

**Comment by jjw24** [comment]
Essentially want to construct two results of different actions:

```
def query(self, query):
        return [
            {
                "Title": "Hello World, this is where title goes. {}".format(('Your query is: ' + query , query)[query == '']),
                "SubTitle": "This is where your subtitle goes",
                "IcoPath": "Images/app.png",
                 "method"="action",
            },
             {
                "Title": "Hello World, this is where title goes. {}".format(('Your query is: ' + query , query)[query == '']),
                "SubTitle": "This is where your subtitle goes",
                "IcoPath": "Images/app.png",
                "method"="action2",
            }
        ]

def action(self, q):
    pass

def action2(self, q):
    pass
```

@Garulf any tips you can think of to help @photkey with this?

**Reply by photkey** [reply]
Yes, that's exactly what I need for an example.`"method"="action"` This should be a typo, I changed it to `"method": "action"`. However, I wrote the simplest test and it doesn't execute the corresponding action, what's wrong with it?
![flow](https://user-images.githubusercontent.com/60434581/144244012-a215a29a-1074-4a0b-851a-6a97892ef3e4.gif)
![Snipaste_2021-12-01_21-34-43](https://user-images.githubusercontent.com/60434581/144244059-7ccec116-3e6f-4f6d-ab79-c24594557392.png)

**Reply by Garulf** [reply]
`method` needs to be inside a dict named `JsonRPCAction`. 

```
"Title": "Title",
"SubTitle": "SubTitle",
"IcoPath": "IcoPath",
"JsonRPCAction": {
    "method": "name_of_method",
    "parameters": [param1, param2]
}
```

**Reply by photkey** [reply]
Thank you very, very much! Finally, I can write my own plugin!

**Reply by Garulf** [reply]
Awesome! Glad to see it works!

If you ever need help feel free to ask.
